components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridPAPStatusDataHandlerTest.java

```diff
@@ -0,0 +1,179 @@
+                registrySimplePAPStatusDataHandler.getStatusData(ABOUT_POLICY, POLICY_KEY, ADD_POLICY, "*");
+        assertEquals(statusDataFromRegistry.length, 0);
+
+        papStatusDataHandler.handle(ABOUT_POLICY, POLICY_KEY, policyStatusHolderList2);
+        statusDataFromDb = jdbcSimplePAPStatusDataHandler.getStatusData(ABOUT_POLICY, POLICY_KEY, GET_POLICY, "*");
+        assertEquals(statusDataFromDb.length, 1);
+        statusDataFromRegistry =
+                registrySimplePAPStatusDataHandler.getStatusData(ABOUT_POLICY, POLICY_KEY, GET_POLICY, "*");
+        assertEquals(statusDataFromRegistry.length, 0);
+
+        StatusHolder[] allStatusDataFromDb =
+                jdbcSimplePAPStatusDataHandler.getStatusData(ABOUT_POLICY, POLICY_KEY, null, "*");
+        assertEquals(allStatusDataFromDb.length, 2);
+        StatusHolder[] allStatusDataFromRegistry =
+                registrySimplePAPStatusDataHandler.getStatusData(ABOUT_POLICY, POLICY_KEY, null, "*");
+        assertEquals(allStatusDataFromRegistry.length, 0);
+    }
+
+    @Test(priority = 10)
+    public void testHandleStatusWhenStatusExistsInDb() throws Exception {
+
+        jdbcSimplePAPStatusDataHandler.handle(ABOUT_POLICY, POLICY_KEY, policyStatusHolderList1);
+        papStatusDataHandler.handle(ABOUT_POLICY, POLICY_KEY, policyStatusHolderList2);
+
+        StatusHolder[] allStatusDataFromDb =
+                jdbcSimplePAPStatusDataHandler.getStatusData(ABOUT_POLICY, POLICY_KEY, null, "*");
+        assertEquals(allStatusDataFromDb.length, 2);
+        StatusHolder[] allStatusDataFromRegistry =
+                registrySimplePAPStatusDataHandler.getStatusData(ABOUT_POLICY, POLICY_KEY, null, "*");
+        assertEquals(allStatusDataFromRegistry.length, 0);
+    }
+
+    @Test(priority = 11)
+    public void testHandleStatusWhenStatusExistsInRegistry() throws Exception {
+
+        registrySimplePAPStatusDataHandler.handle(ABOUT_POLICY, POLICY_KEY, policyStatusHolderList1);
+        papStatusDataHandler.handle(ABOUT_POLICY, POLICY_KEY, policyStatusHolderList2);
+
+        StatusHolder[] allStatusDataFromRegistry =
+                registrySimplePAPStatusDataHandler.getStatusData(ABOUT_POLICY, POLICY_KEY, null, "*");
+        assertEquals(allStatusDataFromRegistry.length, 2);
+        StatusHolder[] allStatusDataFromDb =
+                jdbcSimplePAPStatusDataHandler.getStatusData(ABOUT_POLICY, POLICY_KEY, null, "*");
+        assertEquals(allStatusDataFromDb.length, 0);
+    }
+
+    @Test(priority = 12)
+    public void testGetStatusWhenPolicyStatusExistsInDb() throws Exception {
+
+        jdbcSimplePAPStatusDataHandler.handle(ABOUT_POLICY, POLICY_KEY, policyStatusHolderList1);
+        verifyStatusDataFromStorage();
+    }
+
+    @Test(priority = 13)
+    public void testGetStatusWhenPolicyStatusExistsInRegistry() throws Exception {
+
+        jdbcSimplePAPStatusDataHandler.handle(ABOUT_POLICY, POLICY_KEY, policyStatusHolderList1);
```

Comment:
Should this be `registrySimplePAPStatusDataHandler`?

Reply:
Adressed with https://github.com/wso2/carbon-identity-framework/compare/e393b967246d88953a960b77a0c4a7c355fd9dfd..f26be27fb80abcc3adb9aace8db0909c0eb3c17d